azure-ts-dynamicresource/cdnCustomDomain.ts

```diff
@@ -0,0 +1,258 @@
+// Copyright 2016-2019, Pulumi Corporation.  All rights reserved.
+
+import * as pulumi from "@pulumi/pulumi";
+import * as azure from "@pulumi/azure";
+
+import * as msRestAzure from "@azure/ms-rest-nodeauth";
+import * as cdnManagement from "@azure/arm-cdn";
+import { ServiceClientCredentials } from "@azure/ms-rest-js";
+
+/**
+ * CustomDomainOptions represents the inputs to the dynamic resource.
+ * Any property of type `Input<T>` will automatically be resolved by the custom dynamic resource
+ * before passing them to the dynamic resource provider.
+ */
+export interface CustomDomainOptions {
+    resourceGroupName: pulumi.Input<string>;
+    profileName: pulumi.Input<string>;
+    endpointName: pulumi.Input<string>;
+    customDomainHostName: pulumi.Input<string>;
+    httpsEnabled: boolean;
```

Comment:
`pulumi.Input<boolean>`?

Reply:
I intentionally left it since its value is derived from a pulumi config property. Would you still recommend that I have it as an `Input<boolean>`? How about a union type `Input<boolean> | boolean`?

Reply:
`Input` is itself a union: `Input<T> = T | Promise<T> | OutputInstance<T>` so `Input<boolean>` includes `boolean`.

Reply:
Ah yes I see it now. I've changed it.